django/db/views.py

```diff
@@ -2111,6 +2123,16 @@ def small_molecules_for_dataset(self,dataset_id):
 #        # view of the small molecule table
 #        logger.info(str(('small molecules for dataset',dataset_id,len(queryset))))
 #        return queryset
+
+    # Need a second distinction, because we'll only show the sm column 
+    # if there are datarecord entries with cells
+    def has_small_molecules_for_datarecords(self, dataset_id):
+        sql = ( 'SELECT count (smallmolecule_id) FROM db_datarecord dr '
```

Comment:
The most efficient way of doing this is to use EXISTS:

```sql
select exists(
    select 1 from db_datarecord dr
    where dr.dataset_id=%s and dr.smallmolecule_id is not null
)
```

Your original version needs to actually iterate over all the matching rows to calculate the COUNT, which is basically O(n) where n is the number of rows for that dataset. My version bails out as soon as it finds the first matching row, which is basically O(1). The difference in real terms on our actual data is on the order of a

Note that the choice of `select 1` is arbitrary. The actual expression there should be ignored by the DBMS per the SQL spec (because of the EXISTS wrapped around the SELECT), so it doesn't matter if you write `*` or anything else there. A lot of people like to use `1` just to make it a bit more obvious what the author's intent is.